modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -73,6 +80,53 @@ const AddFDSFieldsModalContent = ({
 		);
 	};
 
+	const synchronizeFDSFields = async () => {
+		const creationData: Array<{name: string; type: string}> = [];
+		const deletionIds: Array<number> = [];
+
+		fields?.forEach((field) => {
+			if (field.selected && !field.id) {
+				creationData.push({name: field.name, type: field.type});
+			}
+
+			if (!field.selected && field.id) {
+				deletionIds.push(field.id);
+			}
+		});
+
+		const formData = new FormData();
+
+		formData.append(
+			`${namespace}creationData`,
+			JSON.stringify(creationData)
+		);
+
+		deletionIds.forEach((id) => {
+			formData.append(`${namespace}deletionIds`, String(id));
+		});
+
+		formData.append(`${namespace}fdsViewId`, fdsView.id);
+
+		const response = await fetch(synchronizeFDSFieldsURL, {
+			body: formData,
+			method: 'POST',
+		});
+
+		if (response.ok) {
+			closeModal();
+
+			onSave();
```

Comment:
I think it might be worth passing the response from the POST to this callback, that way we shouldn't need to call "getFields" again